src/glue.c

```diff
@@ -349,7 +489,113 @@ mg_map *py_dict_to_mg_map(PyObject *dict) {
   return NULL;
 }
 
+// Return 0 on failure
+// Return 1 on success
+int days_since_unix_epoch(int y, int m, int d, int64_t *result) {
+  SCOPED_CLEANUP PyObject *unix_epoch =
+      make_py_datetime(1970, 1, 1, 0, 0, 0, 0);
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(unix_epoch, 0)
+  SCOPED_CLEANUP PyObject *date = make_py_datetime(y, m, d, 0, 0, 0, 0);
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(date, 0)
+  SCOPED_CLEANUP PyObject *method_name = PyUnicode_FromString("__sub__");
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(method_name, 0)
+  SCOPED_CLEANUP PyObject *delta =
+      PyObject_CallMethodObjArgs(date, method_name, unix_epoch, NULL);
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(delta, 0)
+  SCOPED_CLEANUP PyObject *days = PyObject_GetAttrString(delta, "days");
+  IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(days, 0)
+  *result = PyLong_AsLong(days);
+  return 1;
+}
```

Comment:
It feels strange to call python for this. But I couldn't figure out how it can be done in C.

Reply:
Yes because it's so much easier to do this in Python even if you have to go through the C wrappers. One of the reasons I was talking about WASM offline ;)